Sources/Snowplow/VisionOS/Entities/ImmersiveSpaceEntity.swift

```diff
@@ -0,0 +1,144 @@
+//  Copyright (c) 2013-2023 Snowplow Analytics Ltd. All rights reserved.
+//
+//  This program is licensed to you under the Apache License Version 2.0,
+//  and you may not use this file except in compliance with the Apache License
+//  Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+//  http://www.apache.org/licenses/LICENSE-2.0.
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the Apache License Version 2.0 is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+//  express or implied. See the Apache License Version 2.0 for the specific
+//  language governing permissions and limitations there under.
+
+import Foundation
+
+/// The style of a VisionOS immersive space.
+@objc(SPImmersionStyle)
+public enum ImmersionStyle: Int {
+    /// Default immersion style.
+    case automatic
+    /// Displays unbounded content that obscures pass-through video.
+    case full
+    /// Displays unbounded content intermixed with other app content.
+    case mixed
+    /// Content displays with no clipping boundaries applied.
+    case progressive
+}
+
+extension ImmersionStyle {
+    var value: String {
+        switch self {
+        case .automatic:
+            return "automatic"
+        case .full:
+            return "full"
+        case .mixed:
+            return "mixed"
+        case .progressive:
+            return "progressive"
+        }
+    }
+}
+
+/// The visibility of the user's upper limbs in a VisionOS immersive space.
+@objc(SPUpperLimbVisibility)
+public enum UpperLimbVisibility: Int {
+    /// Limbs may be visible or hidden depending on the policies of the component accepting the visibility configuration.
+    case automatic
+    /// Limbs may be visible.
+    case visible
+    /// Limbs may be hidden.
+    case hidden
+}
+
+extension UpperLimbVisibility {
+    var value: String {
+        switch self {
+        case .automatic:
+            return "automatic"
+        case .visible:
+            return "visible"
+        case .hidden:
+            return "hidden"
+        }
+    }
+}
+
+/**
+ Properties for the VisionOS immersive space entity.
+ Entity schema: `iglu:com.apple.swiftui/immersive_space/jsonschema/1-0-0`
+ */
+@objc(SPImmersiveSpaceEntity)
+public class ImmersiveSpaceEntity: SelfDescribingJson {
```

Comment:
Could we make tracking the immersive space work the same as tracking the screen view works? I mean that when you track a screen view event, all events following that will have the screen context entity attached automatically with the ID of the screen view. This is implemented by the `ScreenStateMachine` and can optionally be disabled in the tracker configuration.

I think it would be very useful if starting from the `open_immersive_space` event, all events until the next `open_immersive_space` or until the next `dismiss_immersive_space` event had the immersive space entity automatically attached. I think that should be a matter of copying the `ScreenStateMachine` and adding the option to disable it in the tracker configuration.

Unfortunately, we can't do the same for the windows since there can be multiple open at the same time and we don't know which one the event belongs to.